File: backend/utils/roles.py
from __future__ import annotations

from typing import Iterable

STUDENT_ALIASES = {"student2", "student3"}


def normalize_role_code(code: str | None) -> str:
    normalized = (code or "").lower()
    if normalized in STUDENT_ALIASES:
        return "student"
    return normalized


def normalize_role_set(roles: Iterable[str] | None) -> set[str]:
    if not roles:
        return set()
    return {normalize_role_code(role) for role in roles if role}


def is_student_role(code: str | None) -> bool:
    return normalize_role_code(code) == "student"

File: frontend/src/lib/roles.ts
import type { Role } from "@/lib/types";

const STUDENT_ALIASES = new Set(["student2", "student3"]);

export const normalizeRoleCode = (code?: string | null) => {
  const normalized = (code ?? "").toLowerCase();
  return STUDENT_ALIASES.has(normalized) ? "student" : normalized;
};

export const isStudentRole = (code?: string | null) => normalizeRoleCode(code) === "student";

export const getRoleLabel = (role?: Role | null) => {
  const normalized = normalizeRoleCode(role?.code ?? "");
  if (normalized === "student") return "Ученик";
  return role?.name || "Без роли";
};
